Mathlib/CategoryTheory/Limits/Constructions/Over/Products.lean

```diff
@@ -20,11 +23,97 @@ open CategoryTheory CategoryTheory.Limits
 
 variable {J : Type w}
 variable {C : Type u} [Category.{v} C]
-variable {X : C}
+variable {X Y Z : C}
+
+/-!
+## Binary products
```

Comment:
Can you merge this section with the `BinaryProduct` section below and use the new results to golf `isPullback_of_binaryFan_isLimit`?